### PR TITLE
Small improvements in MulticallProvider

### PR DESCRIPTION
**Pull request by Hyodar**
Noticed some small improvements that could be made in MulticallProvider while casually taking a quick look, so submitting a PR. Most of the improvements are related to typing - now it's considerably better.

### CODE REVIEW

**Review by RCantu92 — APPROVED** [review]
LGTM

File: src/utils/multicall.provider.ts [hunk 1]
```diff
@@ -28,23 +27,30 @@ let multicall2Addresses: Record<number, string> = {
...
+      success: true,
+      returns: {
+        [K in keyof T]: T[K];
+      }
+    ]
+  | [success: false, returns: never[][]];
 
 type GroupTryAllResult<T extends any[][]> = {
-  [K in keyof T]: K extends number ? TryAllResult<T[K]> : T[K];
+  [K in keyof T]: TryAllResult<T[K]>;
```
Comment (0xtaf) [inline src/utils/multicall.provider.ts:53]: this change somehow fails my `npm test` 
I'm assuming it works for you both @RCantu92 @Vxatz ?

Comment (0xtaf) [reply src/utils/multicall.provider.ts:53]: this was fixed after updating the ts version in package.json

**Review by 0xtaf — APPROVED** [review]

**Review by Vxatz — APPROVED** [review]

**Review by RCantu92 — APPROVED** [review]
Thanks for resolving that issue @0xtaf. Approved ✅

**Pull request merged**